app_feup/lib/view/Widgets/form_text_field.dart

```diff
@@ -57,9 +64,14 @@ class FormTextField extends StatelessWidget {
               controller: controller,
               validator: (value) {
                 if (value.isEmpty) {
-                  return emptyText;
+                  return isOptional ? null : emptyText;
+                }
+                switch (validatorType) {
+                  case FormTextFieldValidator.email:
+                    return EmailValidator.validate(value) ? null : emptyText;
+                  default:
+                    return null;
```

Comment:
I think it would be best if the validator function itself was passed to the argument and this function would just become a check if the function is present or not. Wdyt?

Reply:
I'm thinking of a hybrid solution, maybe use a default validator that handles the optional situation, and make it call another function that handles the formatting stuff which was passed by argument (defaults to null). Wdyt?

Reply:
Sounds good!